app/src/main/java/org/p2p/wallet/jupiter/interactor/JupiterSwapInteractor.kt

```diff
@@ -165,19 +165,23 @@ class JupiterSwapInteractor(
     }
 
     /**
-     * We need to cast BigIntegers to Double to keep numbers after dot when dividing.
-     * We had an error when we divided 2039280 / 955187 and got only 2 instead of full 2.13
-     * which caused invalid behaviour in case of slippage 2.1%
+     * Check if price impact is high due to high fees and exceeds given slippage
+     * @param keyAppFeeLamportsString key app fee in lamports (string as it is in json)
+     * @param outAmountLamports out amount in lamports BigInteger
+     * @return [SwapPriceImpactType.HighFees] if price impact is more than given slippage,
+     * [SwapPriceImpactType.None] otherwise
      */
     private fun checkForHighFees(
-        totalFeesAndDeposit: BigInteger,
-        outAmount: BigInteger,
-        slippageValuePercent: Int
-    ): Boolean {
-        val feesValuePercent: Int =
-            (totalFeesAndDeposit.toDouble() / outAmount.toDouble())
-                .times(100)
-                .toInt()
-        return feesValuePercent <= slippageValuePercent
+        keyAppFeeLamportsString: String,
```

Comment:
you can cast keyAppFeeLamportsString in domain entity when mapping

Reply:
changed the type from String to BigInteger and renamed `keyAppFee` to `keyAppFeeInLamports`